Add backend news endpoint

alikaradogan opened the pull request:
## What does this PR do?

* Adds news endpoint.

Commit:
249: add posts endpoint

	new file:   app/routes/posts/posts-router.js

gurolcay commented:
![](https://github.trello.services/images/mini-trello-icon.png) [posts endpointlerinin hazirlanmasi](https://trello.com/c/ihNgDToC/249-posts-endpointlerinin-hazirlanmasi)

alikaradogan commented:
![](https://github.trello.services/images/mini-trello-icon.png) [posts endpointinin news olarak guncellenmesi](https://trello.com/c/6ZwWG4lK/233-posts-endpointinin-news-olarak-guncellenmesi)

alikaradogan commented:
![](https://github.trello.services/images/mini-trello-icon.png) [mocha ile news backend testlerinin eklenmesi](https://trello.com/c/EIDrXsLT/237-mocha-ile-news-backend-testlerinin-eklenmesi)

Commit:
posts: Setup test environment and create unit tests

What is changed:
- Unit tests for posts controller added.
- Test script created for easy test run. (Mocha test framework is required.)
- Axios added to dev dependencies.
- Sequelize logs are disabled for test and production modes.
- Post create method updated. Now it returns created post as a result.
- Post update method fixed. Replace description with body.

Because:
- Now one can just type "npm test" and run all the tests.
- Axios made it a lot easier to make CRUD requests backend.
- Logs were all over the console during tests.
- Create was returning the data sent by user. Now it returns the created post with id.
- Update data is now sync with Post model. Description is not defined on post model, body is.

Remarks:
Mocha can be installed via npm.

It is recommended to change environment to test before running tests.
Otherwise it is not really possible to see the results beacuse of Sequelize logs.

Commit:
Merge branch 'development' of github.com:gtubt/gtubt-backend into 249-posts-endpoint

Commit:
news: Rename posts as news

What is changed:
- Posts renamed as News.

Because:
- Posts name was causing some confusions. News is a better definition.

Commit:
news: Add validation for dates

What is changed:
- Regex added for start end end dates.
- Validation added for start date to be not earlier than current day.
- Validation added for end date to be not earlier than start day.

Because:
- Dates should have a valid string.
- Start date of a news should not be earlier than current day.
- End date of a news should not be earlier than start date.

Remarks:
Validations only added to model but not migration.
Sequelize does not validate entries already in database.

gurolcay commented on test/controllers/test-news-controller.js:
```diff
@@ -0,0 +1,224 @@
+var new_endDate = "2022-04-22";
+
+//Tests
+describe("GTUBT-Backend", function () {
```
en distaki describe'a ihtiyacimiz yok.

alikaradogan replied:
Bir sonraki committe silinecek.

gurolcay commented on test/controllers/test-news-controller.js:
```diff
@@ -0,0 +1,224 @@
+//Tests
+describe("GTUBT-Backend", function () {
+  describe("News Endpoint", function () {
+    describe("Create", function () {
```
create yerine `POST news` seklinde isimlendirme yapalim. digerleri de bunun gibi olsun

alikaradogan replied:
Bir sonraki committe degisecek.

gurolcay commented on test/controllers/test-news-controller.js:
```diff
@@ -0,0 +1,224 @@
+  describe("News Endpoint", function () {
+    describe("Create", function () {
+      before(function (done) {
+        axios
```
chai kullanarak istek atalim. burada ayakta bir server'a ihtiyac duyuyoruz. boyle olmamali

alikaradogan replied:
Bir sonraki committe chai eklenecek ve testler chai'ye gore degisecek.

gurolcay commented on app/models/news/news.js:
```diff
@@ -0,0 +1,96 @@
+          is: /^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/,
+          isDate: true,
+          is_earlier_than_today(startDate) {
+            var d = new Date(),
```
asagidaki sekilde yapsak daha iyi olmaz mi?
```
var d = new Date(startDate)
var today = new Date.now()
if (today > d){
...
}
```

alikaradogan replied:
Olur. 👍🏻

gurolcay commented on app/models/news/news.js:
```diff
@@ -0,0 +1,96 @@
+          is: /^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/,
+          isDate: true,
+          is_earlier_than_start_day(endDate) {
+            if (endDate < this.startDate) {
```
harika!

gurolcay commented on test/controllers/test-news-controller.js:
```diff
@@ -0,0 +1,224 @@
+const assert = require("assert");
```
dosya basinda ```process.env.NODE_ENV = 'test';``` tanimlamasi yapar misin

Commit:
news: Simplify startDate validation

Commit:
Merge branch 'development' of github.com:gtubt/gtubt-backend into 249-posts-endpoint

Commit:
news: Add chai supported tests

What is changed:
- Chai library is now added to the project.
- Axios package is removed from the library.

Because:
- Chai provides easier and more reliable testruns.
- Axios is not necessary anymore since Chai itself supports HTTP calls.

Commit:
news: Use proper HTTP call names for tests.

gurolcay commented on test/controllers/test-news-controller.js:
```diff
@@ -0,0 +1,141 @@
+        .request(server)
+        .get("/api/v1/news/all/")
+        .end((err, res) => {
+          console.log(err);
```
console.log ne icin var?

gurolcay commented on test/controllers/test-news-controller.js:
```diff
@@ -0,0 +1,141 @@
+          done();
+        })
+        .catch((err) => {
+          console.error("Error: " + err.message);
```
error'u bu sekilde basmak yerine throw ettigimiz error icine vermemiz daha dogru olmaz mi? bilmedigim bir sey mi var?

Commit:
news: Remove console logs.

Commit:
news: Refactor tests for news endpoint.

Commit:
news: Refactor handler
